### PR TITLE
Add support for LilyGo T-Display S3

### PR DESCRIPTION
**Pull request by hemna**
This is a first attempt at support for the new LilyGo T-Display S3 board which is an ESP32 S3 device.  The micropython support for the board is here:
https://github.com/Xinyuan-LilyGO/lilygo-micropython/issues/16

### CODE REVIEW

File: drivers/st7789/st7789_8bit_parallel.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+# Tested displays:
+# LilyGo T-display S3
+# https://www.lilygo.cc/products/t-display-s3
+
+import framebuf
+import gc
+import uasyncio as asyncio
+from drivers.boolpalette import BoolPalette
+import st7789
```
Comment (peterhinch) [inline drivers/st7789/st7789_8bit_parallel.py:14]: What is this file?

**Review by peterhinch — COMMENTED** [review]
Is this driver tested?

**Comment by peterhinch** [comment]
The buffer size for a 320x170 display using RGB565 is 108,800 bytes which is excessive. If I were to write a driver for this module I would use 4-bit color with on-the-fly expansion as per the existing ST7789 driver. The nano-gui and micro-gui libraries do not need a full 16 bit color space.

**Comment by peterhinch** [comment]
The LilyGo driver uses a [C module](https://github.com/liangyingy/lcd_binding_micropython/blob/master/lcd/i80_panel.c) to drive the parallel interface.

It is worth considering the performance implications of an 8-bit parallel interface. The SPI used in my driver runs at 30MHz, i.e. 33ns/bit or 264ns/byte. This speed is achieved because SPI is implemented in hardware. It is possible that LilyGo can achieve 264ns/byte with their C interface. It certainly could not be done in Python. About six years ago, with Robert Hammelrath, we developed a GUI for an 8-bit parallel display in Python (I wrote the GUI, he wrote the drivers). We achieved acceptable performance, but Robert went to extraordinary lengths to optimise the Python code, as can be seen [here](https://github.com/peterhinch/micropython-tft-gui/blob/master/tft/driver/TFT_io.py). This involved using Viper code and accessing hardware registers directly.

My policy on drivers for this repo is that they should not require users to recompile the firmware. All current drivers are written in Python. While it might be possible to use techniques such as those described above to write a Python driver it would require understanding the ESP32 at register level. Expertise I don't currently possess. All to achieve a level of performance below that of 30MHz SPI.

In short I don't believe this hardware is a good candidate for a Python driver.

Feel free to continue developing your own driver in Python and C. If you also succeed in developing a performant 4-bit driver in Python, by all means submit it as a new PR and I will consider it. I'm also happy to offer general advice (in Discussions, please). However I'm afraid this PR is a non-starter.
